stl/inc/ximmortalize.h

```diff
@@ -0,0 +1,77 @@
+// ximmortalize.h internal header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _XIMMORTALIZE_H
+#define _XIMMORTALIZE_H
+#include <yvals.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+#ifdef _M_CEE_PURE
+template <class _Ty>
+union _Immortalizer_impl { // constructs _Ty, never destroys
+    _Immortalizer_impl() noexcept : _Storage{} {}
+    _Immortalizer_impl(const _Immortalizer_impl&) = delete;
+    _Immortalizer_impl& operator=(const _Immortalizer_impl&) = delete;
+    ~_Immortalizer_impl() {
+        // do nothing
+    }
+
+    _Ty _Storage;
+};
+
+#pragma warning(push)
+#pragma warning(disable : 4640) // construction of local static object is not thread-safe (/Wall)
+template <class _Ty>
+constexpr _Ty& _Immortalize() noexcept { // return a reference to an object that will live forever

```

Comment:
How can this be `constexpr` when the constructor of `_Immortalizer_impl` is not?

Reply:
More precisely, `_Immortalizer_impl` does not have literal type: it has no `constexpr` non-copy/move constructors, and a non-trivial destructor (In C++20: a non-`constexpr` destructor). 
`constexpr` functions also cannot have variables of static or thread storage duration.

This function isn't even IFNDR, it's straight-up ill-formed.